### DISCUSSION TITLE
✔ CSP against the HTTP response serving it.

### DISCUSSION THREAD

**Discussion by righettod** [oshp/oshp-tracking#25] [Q&A] [answered]
👨‍🎓 **Learning of the day for me:** 

*It seems, unless I missed a subtlety, that a CSP cannot be used to act on the capabilities of a loaded JavaScript script when the CSP is applied on the script itself, via the HTTP response that sends it.*

📺 My test on `Chromium 124.0.6320.0 (Developer Build) (64-bit)`:

https://youtu.be/r_HZJ2lT2SE

![image](https://github.com/oshp/oshp-tracking/assets/1573775/3a9d216f-03ca-4dc9-957c-23cfc1ef736f)

📡 @riramar and the community: Do you have any insights about such behavior?

**Comment by righettod** [comment]
🤔 CSP seems applied for the other CSP directive, like, the `frame-ancestors` one:

![image](https://github.com/oshp/oshp-tracking/assets/1573775/e869d88b-fcfd-4f61-a9ec-e9326d979350)

🤔 Other directives:

![image](https://github.com/oshp/oshp-tracking/assets/1573775/c63fcd8c-10cf-4f81-b160-a6c2950f2641)

**Comment by righettod** [comment]
💡 So the key point I noticed was to apply a CSP **on the page loading** "unsafe" /  "untrusted" resources:

![image](https://github.com/oshp/oshp-tracking/assets/1573775/093b3bc9-7a86-45eb-8e05-cf98916f0a26)

**Comment by riramar** [comment]
Seems strange to me, but I'm not sure if this works as intended.
Sent this question in this group https://lists.w3.org/Archives/Public/public-webappsec/.

**Comment by righettod** [comment]
📚 Thanks a lot for the feedback and for the question. I added the direct ref below:

https://lists.w3.org/Archives/Public/public-webappsec/2024Feb/0006.html

**Comment by noamr** [comment, marked as answer]
Yes, `Content-Security-Policy` is a header that applies to a *document*, not to a sub-resource like a script. This is how it designed...

**Reply by righettod** [reply]
Thank you very much for your help on this matter 👍
